2696. Minimum String Length After Removing Substrings/solution.py

```diff
@@ -1,11 +1,36 @@
 # Minimum String Length After Removing Substrings
 
-# this method takes in a string s with length between 1 and 100 consisting of only uppercase English letters, removes any instances of "AB" or "CD", and returns the length of the resulting sting
+# this method takes in a string s with length between 1 and 100 consisting of only uppercase English letters.
+# it removes any instances of "AB" or "CD", and returns the length of the resulting sting
+# removing one instance of "AB" or "CD" can create a new instance that needs to be removed
 
-# declare a variable strlen for the string length outside of the loop but use it as the limit for the loop
-# loop through the string while incrementing by 1 with strlen - 1 as the ending
-	# inner loop through the whole string, check if each index + the next letter is equal to AB
-	# second inner loop check if each index + the next letter is equal to CD
-		# in both loops, if a match is found, cut those two characters out of the string and reform as a new string
-			# if this happens decrement strlen by 2
-  # return strlen after the loops have finished
+class Solution(object):
+	def minLength(self, s):
+		
+	# declare a variable strlen for the string length outside of the loop and use it as the limit for the while
+	strlen = len(s)
+	
+	# loop through the string
+	i = 0
+	while i < strlen:
+		# check if each index + the next letter is equal to AB
+		if s[i:i+2] == "AB":
+			# in both if statements, if a match is found, cut those two characters out of the string and
+			# reform as a new string
+			s = s[:i] + s[i+2:]
+			# since we cut out characters we decrement the length of the string
+			strlen = strlen - 2
+			# reset i to 0 so that we can start over from the beginning of the string to scan for more instances
+			i = 0
```

Comment:
Do you have to reset back to 0?

Reply:
I think I do - otherwise we would be searching from the index's old value, so if the "i" index was 4 we'd continue to the next while iteration looking in the middle of the string rather than starting over from the beginning, and there might be a new AB or CD before index 4.